client/src/state/secure.rs

```diff
@@ -1,39 +1,42 @@
 // Copyright 2020-2021 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::{
-    actors::{InternalResults, SHRequest, SHResults},
-    line_error,
-    utils::LoadFromPath,
-    Location,
-};
+//! Secure Client Actor State
+
+use crate::line_error;
+
+use crate::{state::key_store::KeyStore, utils::LoadFromPath, Location};
 
 use engine::{
     store::Cache,
-    vault::{ClientId, RecordId, VaultId},
+    vault::{BoxProvider, ClientId, DbView, RecordId, VaultId},
 };
-
-use riker::actors::*;
-
 use std::{collections::HashSet, time::Duration};
 
-use serde::{Deserialize, Serialize};
-
+/// Cache type definition
 pub type Store = Cache<Vec<u8>, Vec<u8>>;
 
-/// A [`Client`] Cache Actor which routes external messages to the rest of the Stronghold system.
-#[actor(SHResults, SHRequest, InternalResults)]
-#[derive(Debug, Serialize, Deserialize, Clone)]
-pub struct Client {
+pub struct SecureClient<Pr>
+where
+    Pr: BoxProvider + Send + Sync + Clone + 'static + Unpin,
```

Comment:
Remove the Generic, instead directly use `internals::Provider`, since this is redundant. The interface, the Registry and the snapshot actor all replace this generic with  `internals::Provider`, thus there is no way to use the `SecureClient` with any other Provider.

Reply:
agreed.